openupgrade_scripts/scripts/sale_timesheet/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,32 @@
+from openupgradelib import openupgrade
+
+
+def _update_account_analytic_line_timesheet_invoice_type(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        WITH timesheet_info as (
+            SELECT aal.id as id,
+                   product_tmpl.service_type as service_type
+            FROM account_analytic_line aal
+            JOIN sale_order_line sol
+             ON sol.id = aal.so_line
+            JOIN product_product product
+             ON product.id = sol.product_id
+            JOIN product_template product_tmpl
+             ON product_tmpl.id = product.product_tmpl_id
+            WHERE product_tmpl.type = 'service'
+                  AND product_tmpl.invoice_policy = 'delivery'
+                  AND product_tmpl.service_type IN ('milestones', 'manual')
```

Comment:
How could `product.template.service_type` could be set to `milestones` since this selection value is new to v16 ?

(openupgrade noob asking)